projects/js-packages/connection/components/in-place-connection/index.jsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useRef } from 'react';
+import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+/**
+ * The in-place connection component.
+ *
+ * @param {object} props -- The properties.
+ * @param {string} props.title -- Element title.
+ * @param {boolean} props.isLoading -- Whether the element is still loading.
+ * @param {string|number} props.width -- Iframe width.
+ * @param {string|number} props.height -- Iframe height.
+ * @param {boolean} props.hasConnectedOwner -- Whether the site has connection owner connected.
+ * @param {boolean} props.scrollToIframe -- Whether we need to auto-scroll the window upon element rendering.
+ * @param {string} props.connectUrl -- The connection URL.
+ * @param {Function} props.onComplete -- The callback to be called upon complete of the connection process.
+ * @param {string} props.source -- Component location identifier passed to WP.com.
+ *
+ * @returns {React.Component} The in-place connection component.
+ */
+const InPlaceConnection = props => {
+	const {
+		title,
+		isLoading,
+		width,
+		hasConnectedOwner,
+		scrollToIframe,
+		connectUrl,
+		onComplete,
+		source,
+	} = props;
+	let { height } = props;
+
+	const iframeWrapRef = useRef();
+	const iframeRef = useRef();
+
+	const receiveData = e => {
+		if ( e.source !== iframeRef.current.contentWindow ) {
+			return;
+		}
+
+		switch ( e.data ) {
+			case 'close':
+				// Remove listener, our job here is done.
+				window.removeEventListener( 'message', receiveData );
+
+				if ( onComplete ) {
+					onComplete();
+				}
+				break;
+			case 'wpcom_nocookie':
+				// Third-party cookies blocked. Let's redirect.
+				window.location.replace( connectUrl );
```

Comment:
How about we make this configurable for our generic component?
eg add an `onThirdPartyCookiesBlocked` prop?

Reply:
Good point, I extracted this code into a helper: https://github.com/Automattic/jetpack/pull/19134/commits/3ec1cbf74745384a087627ef942259b4373cb17a